src/utils/threads/thread_pool.cpp

```diff
@@ -132,12 +132,13 @@ void thread_pool::do_thread_work_()
 void thread_pool::do_thread_work_pinned_thread_(int tid, cpu_set_t cpu_set)
 {
   // Set the CPU affinity for the thread
-  auto error = pthread_setaffinity_np(pthread_self(),
-                                      sizeof(cpu_set_t), &cpu_set);
-  if (error != 0) {
-    std::cerr << "error in pthread_setaffinity_np, error="
-              << error << std::endl;
-  }
+ // auto error = pthread_setaffinity_np(pthread_self(),
+   //                                   sizeof(cpu_set_t), &cpu_set);
+    //if (error != 0) {
+      //  std::cerr << "error in pthread_setaffinity_np, error="
+        //     << error << std::endl;
+        
+    //}
```

Comment:
Can you remove this modification? Or, is it necessary?

Reply:
This was just more of a convenience, when I run on lassen this error prints for each rank. Tom said that is did not matter and that it was an issue of pthreads missing some setting with power9s. I can un-comment it.

Reply:
Yes, please uncomment it and minimize the other changes of this PR. Non essential changes should be recovered.